### Step 1. 필요한 라이브러리 임포트

In [1]:
import pandas as pd

### Step 2. [데이터](https://raw.githubusercontent.com/justmarkham/DAT8/master/data/u.user) 확인

- user_id 유저 아이디
- age 나이
- gender 성별
- occupation 직업
- zip_code 우편번호

### Step 3. 

**read_csv를 통해 데이터를 올바른 형태로 불러온 뒤 users에 저장해주세요.**  
data : occupation.tsv

user_id를 인덱스로 설정해주세요.

In [99]:
users = pd.read_csv('occupation.tsv', sep='|', index_col='user_id')
users

age gender     occupation zip_code
user_id                                    
1         24      M     technician    85711
2         53      F          other    94043
3         23      M         writer    32067
4         24      M     technician    43537
5         33      F          other    15213
...      ...    ...            ...      ...
939       26      F        student    33319
940       32      M  administrator    02215
941       20      M        student    97229
942       48      F      librarian    78209
943       22      M        student    77841

[943 rows x 4 columns]

### Step 4. 초반 25행 조회

In [100]:
users.iloc[:25,:]

age gender     occupation zip_code
user_id                                    
1         24      M     technician    85711
2         53      F          other    94043
3         23      M         writer    32067
4         24      M     technician    43537
5         33      F          other    15213
6         42      M      executive    98101
7         57      M  administrator    91344
8         36      M  administrator    05201
9         29      M        student    01002
10        53      M         lawyer    90703
11        39      F          other    30329
12        28      F          other    06405
13        47      M       educator    29206
14        45      M      scientist    55106
15        49      F       educator    97301
16        21      M  entertainment    10309
17        30      M     programmer    06355
18        35      F          other    37212
19        40      M      librarian    02138
20        42      F      homemaker    95660
21        26      M         writer    30068
22        25      M         writer    40206
23        30      F         artist    48197
24        21      F         artist    94533
25        39      M       engineer    55107

### Step 5. 마지막 10행 조회

In [101]:
users.iloc[-10:,:]
users.tail(10)

age gender     occupation zip_code
user_id                                    
934       61      M       engineer    22902
935       42      M         doctor    66221
936       24      M          other    32789
937       48      M       educator    98072
938       38      F     technician    55038
939       26      F        student    33319
940       32      M  administrator    02215
941       20      M        student    97229
942       48      F      librarian    78209
943       22      M        student    77841

### Step 6. 데이터셋은 몇 개의 행으로 이루어져 있나요?

943

In [102]:
len(users)

(943, 4)

In [103]:
users.shape

(943, 4)

### Step 7.  데이터셋은 몇 개의 열로 이루어져 있나요?

4

In [8]:
len(user_id.columns)

5

In [104]:
users.shape

(943, 4)

### Step 8. 열(column)의 이름들을 모두 조회해주세요

Index(['age', 'gender', 'occupation', 'zip_code'], dtype='object')

In [105]:
users.columns

Index(['age', 'gender', 'occupation', 'zip_code'], dtype='object')

### Step 9. 데이터셋의 인덱스를 조회해주세요

Int64Index([  1,   2,   3,   4,   5,   6,   7,   8,   9,  10,
            ...
            934, 935, 936, 937, 938, 939, 940, 941, 942, 943],
           dtype='int64', name='user_id', length=943)

In [106]:
users.index

Int64Index([  1,   2,   3,   4,   5,   6,   7,   8,   9,  10,
            ...
            934, 935, 936, 937, 938, 939, 940, 941, 942, 943],
           dtype='int64', name='user_id', length=943)

### Step 10. 각 열의 데이터타입은 무엇인가요?

In [107]:
users.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 943 entries, 1 to 943
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   age         943 non-null    int64 
 1   gender      943 non-null    object
 2   occupation  943 non-null    object
 3   zip_code    943 non-null    object
dtypes: int64(1), object(3)
memory usage: 36.8+ KB


In [110]:
users.dtypes

age            int64
gender        object
occupation    object
zip_code      object
dtype: object

### Step 11. occupation column만 조회해주세요

In [108]:
users['occupation']

user_id
1         technician
2              other
3             writer
4         technician
5              other
           ...      
939          student
940    administrator
941          student
942        librarian
943          student
Name: occupation, Length: 943, dtype: object

In [111]:
users.occupation

user_id
1         technician
2              other
3             writer
4         technician
5              other
           ...      
939          student
940    administrator
941          student
942        librarian
943          student
Name: occupation, Length: 943, dtype: object

### Step 12. 데이터셋에는 몇 종류의 직업이 존재하나요?

21

In [117]:
groups = users.groupby('occupation')
len(groups)

21

In [118]:
len(users['occupation'].unique())

21

### Step 13. 데이터셋에서 가장 많이 존재하는 직업은 무엇인가요?

student

In [114]:
users['occupation'].value_counts()

student          196
other            105
educator          95
administrator     79
engineer          67
programmer        66
librarian         51
writer            45
executive         32
scientist         31
artist            28
technician        27
marketing         26
entertainment     18
healthcare        16
retired           14
lawyer            12
salesman          12
none               9
homemaker          7
doctor             7
Name: occupation, dtype: int64

In [123]:
users.groupby('occupation')['age'].count().sort_values(ascending=False)

occupation
student          196
other            105
educator          95
administrator     79
engineer          67
programmer        66
librarian         51
writer            45
executive         32
scientist         31
artist            28
technician        27
marketing         26
entertainment     18
healthcare        16
retired           14
lawyer            12
salesman          12
none               9
homemaker          7
doctor             7
Name: age, dtype: int64

### Step 14. users.describe() 코드를 실행해서 숫자 열의 요약정보를 조회해주세요

In [124]:
users.describe() 

age
count  943.000000
mean    34.051962
std     12.192740
min      7.000000
25%     25.000000
50%     31.000000
75%     43.000000
max     73.000000

### Step 15. user들의 평균 나이는 몇 살인가요?
34

In [125]:
int(user_id['age'].mean())

34

### Step 16. 데이터셋에서 각각 여성과 남성 직업 순위 Top 5를 표시해주세요

* F  
occupation  
student          60  
administrator    36  
other            36  
librarian        29  
educator         26  
* M  
occupation  
student       136  
educator       69  
other          69  
engineer       65  
programmer     60  

In [130]:
users.groupby('gender')['occupation'].value_counts().loc['F'].head()

occupation
student          60
administrator    36
other            36
librarian        29
educator         26
Name: occupation, dtype: int64

In [131]:
users.groupby('gender')['occupation'].value_counts().loc['M'].head()

occupation
student       136
educator       69
other          69
engineer       65
programmer     60
Name: occupation, dtype: int64